doc/docs/master/enterprise/auth/oidc/configure-keycloak.md

```diff
@@ -0,0 +1,256 @@
+   * `password:` `admin`
+
+   ![Keycloak login](../../../assets/images/s_keycloak_login.png)
+
+1. If you do not have one already, add a new realm. Otherwise, proceed to
+the next step.
+
+   A realm is a logically isolated group, similar to a tenant. By default,
+   Keycloak predefines a `Master` realm for administrative purposes. You need
+   to create a new realm for Pachyderm. For example, you can call it
+   `pachyderm`.
+
+1. If you have created a new realm you need to add users to whom you want to
+grant access to Pachyderm. Create a user by clicking **Users > Add user** and
+filling out the required fields. Otherwise, proceed to the next step.
+
+   1. Set up a password for this user by clicking **Credentials** and typing
+   your password information.
+
+      You might want to set the temporary password toggle to off so that your
+      password does not expire.
+
+1. Create a client for Pachyderm:
+
+   1. Click **Clients > Create**.
+   1. Specify a client ID. For example, *pachyderm-test*.
+   1. Set the **Client Protocol** to **openid-connect**.
+   1. Set the **Root URL** to `http://<ip>:30657/authorization-code/callback`.
+      For example:
+
+      ```bash
+      http://192.168.64.36:30657/authorization-code/callback
+      ```
+
+      You can configure additional settings, such as require consent and so on,
+      as needed, but this minimal configuration is everything you need to make
+      Keycloak work with Pachyderm.
+
+      ![Keycloak Client](../../../assets/images/s_keycloak_client.png)
+
+1. Proceed to [Configure Pachyderm](#confgiure-pachyderm).
+
+## Configure Pachyderm
+
+Now that you have Keycloak configured, you need to log in to Pachyderm
+with the user you have created in Keycloak.
+
+To configure Pachyderm, complete the following steps:
+
+1. Go to the terminal and forward the `pachd` pod to the OIDC port:
+
+   1. Get the `pachd` pod ID:
+
+      ```bash
+      kubectl get pod
+      ```
+
+      **Example system response:**
+
+      ```bash
+      dash-5768cb7d98-j6cgt       2/2     Running   0          4h2m
+      etcd-56d897697-xzsqr        1/1     Running   0          4h2m
+      keycloak-857c59449b-htg99   1/1     Running   0          4h6m
+      pachd-79f7f68c65-9qs8g      1/1     Running   0          4h2m
+      ```
+
+   1. Forward the `pachd` pod to the OIDC port:
+
+      **Example:**
+
+      ```bash
+      kubectl port-forward pachd-79f7f68c65-9qs8g 30657
+      ```
+
+1. Enable Pachyderm authentication:
+
+   ```bash
+   pachctl auth activate --initial-admin=robot:admin
+   ```
+
+   Pachyderm returns a token.
+
+   **WARNING!** You must save the token to a secure location
+   to avoid being locked out of your cluster.
+
+1. Log in as the admin user with the token you received in the previous
+step:
+
+   ```bash
+   pachctl auth use-auth-token
+   ```
+
+1. Set up the authentication config:
+
+    ```bash
+    pachctl auth set-config <<EOF
+    {
+            "live_config_version": 2,
```

Comment:
probably should set this to version 1, version 2 only needed if you're updating the config